code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -480,15 +483,43 @@ The force parameter is for messages that should ignore a dead queen
 to_chat will check for valid clients itself already so no need to double check for clients
 
 */
-/datum/hive_status/proc/xeno_message(message = null, size = 3, force = FALSE)
+
+///Used for Hive Message alerts
+/datum/hive_status/proc/xeno_message(message = null, size = 3, force = FALSE, sound = null, atom/target = null, apply_preferences = FALSE, filter_list = null)
 	if(!force && !can_xeno_message())
 		return
-	for(var/i in get_all_xenos())
-		var/mob/living/carbon/xenomorph/X = i
+
+	var/mob/living/carbon/xenomorph/X
+	var/list/final_list = get_all_xenos()
+
+	if(filter_list) //Filter out Xenos in the filter list
+		for(X in filter_list)
+			final_list -= X
+
+	for(X in final_list)
+
 		if(X.stat) // dead/crit cant hear
 			continue
+
+		if(!X.client) // If no client, there's no point; also runtime prevention
+			continue
+
+		if(apply_preferences && X.client.prefs.toggles_alerts & ALERTS_XENO_HEALTH) //If xeno alert preferences matter and we have them turned off, skip
+			continue
```

Comment:
This should be before the proc is called not in the actual proc

Reply:
the xeno messages are global and address all xenos until filtered out at the actual proc; are you saying to add people who have alerts turned off to the filter list in the health alert proc that calls the message? I don't know exactly what you want here